platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/BundleProcessor.java

```diff
@@ -102,92 +127,63 @@ public boolean stopBundle(ProfileMigrationReport report, Bundle bundle) {
    * originally in the corresponding state.
    *
    * @param context the bundle context to use for managing bundles
-   * @param report the report where to record errors
    * @param jprofile the profile where to retrieve the set of bundles from the original system
    * @param tasks the task list where to record tasks to be executed
-   * @return <code>false</code> if we failed to retrieve from memory a left over bundle; <code>
-   *     true</code> if everything was ok
    */
-  public boolean processBundles(
-      BundleContext context, ProfileMigrationReport report, JsonProfile jprofile, TaskList tasks) {
-    LOGGER.debug("Processing bundles import");
-    final Map<String, JsonBundle> jbundlesMap =
-        jprofile
-            .bundles()
-            .collect( // linked map to preserve order
-                org.codice.ddf.admin.application.service.migratable.Collectors.toLinkedMap(
-                    JsonBundle::getFullName, Function.identity()));
+  public void processBundlesToRecordTasks(
+      BundleContext context, JsonProfile jprofile, TaskList tasks) {
+    LOGGER.debug("Processing bundles");
+    final Map<String, Bundle> bundlesMap =
+        Stream.of(listBundles(context))
+            .collect(Collectors.toMap(JsonBundle::getFullName, Function.identity()));
 
-    processMemoryBundles(context, jbundlesMap, tasks);
-    return processLeftoverExportedBundles(context, report, jbundlesMap, tasks);
+    processExportedBundlesToRecordTasks(context, jprofile, bundlesMap, tasks);
+    processLeftoverBundlesToRecordTasks(bundlesMap, tasks);
   }
 
   /**
-   * Processes bundles in memory by recording tasks to start, stop, install, or uninstall bundles
-   * that were originally in the corresponding state.
+   * Processes bundles that were exported by recording tasks to start, stop, install, or uninstall
+   * bundles that were originally in the corresponding state.
    *
    * <p><i>Note:</i> Any bundles found in memory are removed from the provided map since they have
    * been processed.
    *
    * @param context the bundle context to use for managing memory bundles
-   * @param jbundles the set of bundles on the original system
+   * @param jprofile the profile where to retrieve the set of bundles from the original system
+   * @param bundles the set of bundles in memory
    * @param tasks the task list where to record tasks to be executed
    */
-  public void processMemoryBundles(
-      BundleContext context, Map<String, JsonBundle> jbundles, TaskList tasks) {
-    LOGGER.debug("Processing bundles defined in memory");
-    for (final Bundle bundle : listBundles(context)) {
-      final String name = JsonBundle.getFullName(bundle);
-      final JsonBundle jbundle = jbundles.remove(name); // remove from jbundles when we find it
-
-      processBundle(context, jbundle, bundle, tasks);
-    }
+  public void processExportedBundlesToRecordTasks(
```

Comment:
I added the `toRecordTasks` at your request from the other PR. ;-)

Reply:
Hmm... I don't remember and couldn't find where I suggested that name. You did suggest calling them `processXXXXAndPopulateTaskList()` (https://github.com/codice/ddf/pull/2707#discussion_r160749067), which I find much better than `processXXXXToRecordTasks()`. But `recordXXXXTasks()` would obviously be best 😆

Reply:
I'll change it, again, to processXXXXAndPopulateTaskList(). 

I do not believe recordXXXXTasks() is correct since you are not record XXXX but actually processing XXXX which will result in tasks being added to the TaskList. 

These tasks can be anything. For example, processing a feature that was started could result in some case in a task to restart it and in another case in a task to install it first. 

I originally had called these methods only processXXX as all of them are designed to record/populate tasks so I felt like adding that phrase to the method name was simply redundant.